portfolio/src/main/webapp/script.js

```diff
@@ -34,12 +34,57 @@ function addRandomFact() {
  * Fetches JSON String from the server
  */
 function getComments() {
+    console.log("Hello world!");
   fetch('/data').then(response => response.text()).then((comment) => {
     document.getElementById('comment-container').innerText = comment;
   });
 }
 
-function adjust_textarea(h) {
+function adjustTextarea(h) {
     h.style.height = "20px";
     h.style.height = (h.scrollHeight)+"px";
 }
+
+/** Fetches comments from the server and adds them to the DOM. */
+function loadComments() {
+  fetch('/data?num-results=' + getCommentLimit()).then(response => response.json()).then((comments) => {
+    const commentListElement = document.getElementById('comment-list');
+    comments.forEach((comment) => {
+      commentListElement.appendChild(createCommentElement(comment));
+    })
+  });
+}
+
+/** Creates an element that represents a comment through manual string concatenation*/
+function createCommentElement(comment) {
+    const commentElement = document.createElement('li');
+    commentElement.className = 'comment';
+    const nameElement = document.createElement('span');
+
+    var str = comment.name; // name
+    str += " at ";
+    str += comment.email; // email
+    str += " says: ";
+    str += comment.subject; // subject
+    str += " // ";
+    str += comment.comments; // comments
+    nameElement.innerText = str;
+
+    commentElement.appendChild(nameElement);
+    return commentElement;
+}
+
+function getCommentLimit() {
+    let tmp = (new URL(document.location)).searchParams;
+    let res = tmp.get("num-results");
+    if(!res || res.length === 0) {
+        return "10";
+    }
+    return res;
+}
+
+function deleteComments() {
+  const request = new Request('/delete-data', {method: 'POST'});
```

Comment:
Do you want to get user confirmation before sending the delete request to the backed:

https://www.w3schools.com/jsref/met_win_confirm.asp